### PR TITLE
Support 1.10 in the integration tests

### LINKED ISSUE #1623: Include Kubernetes 1.10 in testing platforms
Our CI builds should be updated to include the new release of Kubernetes 1.10.

It would also be nice to update `minikube.sh` so that 1.10 is the new default.

### PR DESCRIPTION
**Pull request by travisn**
Add support for K8s 1.10 in the jenkins integration tests. Given some changes in shutdown behavior, this had been dependent on #1701. A number of test improvements to help troubleshoot #1701 are also included in this PR.

Which issue is resolved by this Pull Request:
Resolves #1623

Checklist:
- [ ] Documentation has been updated, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.

### CODE REVIEW

File: tests/framework/utils/k8s_helper.go [hunk 1]
```diff
@@ -384,6 +363,37 @@ func (k8sh *K8sHelper) WaitUntilPodWithLabelDeleted(label string, namespace stri
...
+	if err != nil {
+		logger.Info("failed to get pod status in namespace %s. %+v", namespace, err)
+		return
+	}
+	for _, pod := range pods.Items {
+		logger.Infof("%s (%s) pod status: %+v", pod.Name, namespace, pod.Status)
+	}
+}
+
+func (k8sh *K8sHelper) PrintPodDescribe(name, namespace string) {
```
Comment (jbw976) [inline tests/framework/utils/k8s_helper.go]: `kubectl` has a describe command, is that not a standard verb available in the k8s API too?  what does the `kubectl` implementation do?

File: tests/framework/utils/k8s_helper.go [hunk 2]
```diff
@@ -528,14 +541,47 @@ func (k8sh *K8sHelper) waitForVolumeAttachment(namespace, volumeAttachmentName s
...
+	for _, pv := range pvs.Items {
+		names = append(names, pv.Name)
+	}
+	logger.Infof("Found PVs: %v", names)
+}
+
+func (k8sh *K8sHelper) printPVCs(namespace string) {
+	pvcs, err := k8sh.Clientset.CoreV1().PersistentVolumeClaims(namespace).List(metav1.ListOptions{})
+	if err != nil {
+		logger.Infof("faied to list pvs. %+v", err)
```
Comment (jbw976) [inline tests/framework/utils/k8s_helper.go]: typo: `faied` (this typo appears multiple places)

also, the resource being listed is PVCs, not PVs in this function.

File: tests/scripts/kubeadm.sh [hunk 3]
```diff
@@ -41,7 +41,7 @@ EOF
     echo "wait for K8s master node to be Ready"
     INC=0
     while [[ $INC -lt 20 ]]; do
-        kube_ready=$(kubectl get node -o jsonpath='{.items[0].status.conditions[3].status}')
+        kube_ready=$(kubectl get node -o jsonpath='{.items['$count'].status.conditions[?(@.reason == "KubeletReady")].status}')
```
Comment (jbw976) [inline tests/scripts/kubeadm.sh]: this is some fancy looking bash here.  is the conditions index number not always the same i take it?

Comment (travisn) [reply tests/scripts/kubeadm.sh]: Previously we relied on index 3, but now it's index 4. Correct, this gets us away from relying on the index.

File: tests/framework/utils/k8s_helper.go (hunk 1 above)
Comment (travisn) [reply tests/framework/utils/k8s_helper.go]: the [kubectl implementation](https://github.com/kubernetes/kubernetes/blob/5782d476c533c1f60c09c09ef940af0f20f59189/pkg/printers/internalversion/describe.go#L205) also gets the object and then prints the events. This isn't quite as elegant, but it was sufficient.

**Pull request merged**